packages/nx/src/tasks-runner/tasks-schedule.ts

```diff
@@ -45,6 +45,15 @@ export class TasksSchedule {
     );
   }
 
+  private getRunningTasks(): string[] {
```

Comment:
You were right before, maintaining the array is cheaper than recalculating it all the time.